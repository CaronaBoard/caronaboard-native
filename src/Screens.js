import { Navigation } from 'react-native-navigation'
import { Provider } from 'react-redux'

import SignInScreen from './Containers/SignInScreen'
import RideList from './Containers/RideList'
import TestScreen from './Containers/TestScreen'

export default function (store) {
  Navigation.registerComponent('carona.signIn', () => SignInScreen, store, Provider)
  Navigation.registerComponent('carona.rideList', () => RideList, store, Provider)
  Navigation.registerComponent('carona.testScreen', () => TestScreen, store, Provider)
}
