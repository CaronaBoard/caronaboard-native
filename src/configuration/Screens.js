import { Navigation } from 'react-native-navigation'
import { Provider } from 'react-redux'

import SignInScreen from '../modules/authentication/containers/LoginScreen'
import SignUpScreen from '../modules/authentication/containers/SignUpScreen'
import ProfileScreen from '../modules/authentication/containers/ProfileScreen'
import RideList from '../modules/rideRequest/containers/RideList'
import RideOffer from '../modules/rideRequest/containers/RideOfferScreen'
import RideRequest from '../modules/rideRequest/containers/RideRequestScreen'

const registerScreens = (store) => {
  Navigation.registerComponent('carona.signIn', () => SignInScreen, store, Provider)
  Navigation.registerComponent('carona.signUp', () => SignUpScreen, store, Provider)
  Navigation.registerComponent('carona.profile', () => ProfileScreen, store, Provider)
  Navigation.registerComponent('carona.rideList', () => RideList, store, Provider)
  Navigation.registerComponent('carona.rideOffer', () => RideOffer, store, Provider)
  Navigation.registerComponent('carona.rideRequest', () => RideRequest, store, Provider)
}

export default {
  registerScreens
}
