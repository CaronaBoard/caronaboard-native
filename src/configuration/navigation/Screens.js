import SignInScreen from '../../modules/authentication/containers/SignInScreen'
import SignUpScreen from '../../modules/authentication/containers/SignUpScreen'
import ForgotPasswordScreen from '../../modules/authentication/containers/ForgotPasswordScreen'
import ProfileScreen from '../../modules/authentication/containers/ProfileScreen'
import RideList from '../../modules/rideRequest/containers/RideList'
import RideOffer from '../../modules/rideRequest/containers/RideOfferScreen'
import RideRequest from '../../modules/rideRequest/containers/RideRequestScreen'

export const screens = {
  signIn: {
    id: 'authentication.signIn',
    component: SignInScreen
  },
  signUp: {
    id: 'authentication.signUp',
    component: SignUpScreen
  },
  forgotPassword: {
    id: 'authentication.forgotPassword',
    component: ForgotPasswordScreen
  },
  profile: {
    id: 'user.profile',
    component: ProfileScreen,
    title: 'Profile'
  },
  rideList: {
    id: 'ride.list',
    component: RideList,
    label: 'Ride List',
    title: 'Ride List'
  },
  rideOffer: {
    id: 'ride.offer',
    component: RideOffer,
    label: 'Ride Offer',
    title: 'Ride Offer'
  },
  rideRequest: {
    id: 'ride.request',
    component: RideRequest,
    label: 'Ride Request',
    title: 'Ride Request'
  }
}

export const tabs = [
  {
    label: screens.rideList.label,
    screen: screens.rideList.id,
    title: screens.rideList.title,
    icon: require('../../assets/images/swap.png')
  }, {
    label: screens.rideOffer.label,
    screen: screens.rideOffer.id,
    title: screens.profile.title,
    icon: require('../../assets/images/swap.png')
  }
]
