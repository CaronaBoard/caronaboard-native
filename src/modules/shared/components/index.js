export { Button } from './Button'
export { TextInput } from './TextInput'
export { GradientButton } from './GradientButton'
export { LinearGradient } from './LinearGradient'
