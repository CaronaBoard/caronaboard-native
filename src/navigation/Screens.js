import SignUpScreen from '../modules/authentication/containers/SignUpScreen'
import ForgotPasswordScreen from '../modules/authentication/containers/ForgotPasswordScreen'
import ProfileScreen from '../modules/authentication/containers/ProfileScreen'
import RideList from '../modules/rideOffer/containers/RideOffersListContainer'
import RideOffer from '../modules/rideOffer/containers/NewRideOfferContainer'
import RideRequest from '../modules/rideRequest/containers/NewRideRequestContainer'
import YourRideOfferList from '../modules/rideOffer/containers/YourRideOffersScreen'
import YourRideRequestsScreen from '../modules/rideRequest/containers/YourRideRequestsScreen'
import SignInScreen from '../modules/authentication/containers/SignInScreen'
import Settings from '../modules/settings/containers/SettingsContainer'
import { Feedback } from '../modules/settings/components/Feedback'

export const screens = {
  signIn: {
    id: 'authentication.signIn',
    component: SignInScreen
  },
  signUp: {
    id: 'authentication.signUp',
    component: SignUpScreen
  },
  forgotPassword: {
    id: 'authentication.forgotPassword',
    component: ForgotPasswordScreen
  },
  rideOfferList: {
    id: 'ride.list',
    component: RideList,
    label: 'Offer List',
    title: 'Offer List'
  },
  rideOffer: {
    id: 'ride.offer',
    component: RideOffer,
    label: 'Ride Offers',
    title: 'Ride Offers'
  },
  rideRequest: {
    id: 'ride.request',
    component: RideRequest,
    label: 'Ride Request',
    title: 'Ride Request'
  },
  yourRideOffersList: {
    id: 'ride.offer.list',
    component: YourRideOfferList,
    label: 'Your Ride Offers',
    title: 'Your Ride Offers'
  },
  yourRequestsList: {
    id: 'ride.request.list',
    component: YourRideRequestsScreen,
    label: 'Your Ride Requests',
    title: 'Your Ride Requests'
  },
  settings: {
    id: 'settings',
    component: Settings,
    title: 'Settings'
  },
  profile: {
    id: 'settings.profile',
    component: ProfileScreen,
    title: 'Profile'
  },
  feedback: {
    id: 'settings.feeback',
    title: 'Feedback',
    component: Feedback
  }
}
