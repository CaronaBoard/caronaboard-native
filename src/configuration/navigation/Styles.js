import { Colors, Fonts } from '../../modules/shared/themes'

export const style = {
  tabBarBackgroundColor: Colors.blue,
  navBarButtonColor: Colors.white,
  tabBarButtonColor: Colors.white,
  navBarTextColor: Colors.white,
  tabBarSelectedButtonColor: Colors.darkblue,
  navigationBarColor: Colors.blue,
  navBarBackgroundColor: Colors.blue,
  statusBarColor: Colors.bloodOrange,
  tabFontFamily: Fonts.type.base
}

export const barsHidden = {
  navBarHidden: true,
  tabBarHidden: true,
  statusBarHidden: true,
  drawUnderNavBar: true,
  drawUnderTabBar: true
}
