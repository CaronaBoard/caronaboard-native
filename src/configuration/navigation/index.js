import { Navigation } from 'react-native-navigation'
import { Provider } from 'react-redux'

import { screens, tabs } from './Screens'
import { style, barsHidden } from './Styles'

const { startTabBasedApp, startSingleScreenApp, registerComponent } = Navigation

const tabAppParams = {
  tabs,
  animationType: 'fade',
  tabsStyle: style,
  appStyle: style
}

const singleAppParams = {
  screen: {
    screen: 'authentication.signIn',
    navigatorStyle: barsHidden
  }
}

const isUserSignedIn = () => true
const startCaronaBoard = () => startTabBasedApp(tabAppParams)
const startAuthentication = () => startSingleScreenApp(singleAppParams)

const startApp = () => {
  isUserSignedIn() ? startCaronaBoard() : startAuthentication()
}

const registerScreens = (store) => {
  for (const screen in screens) {
    const { id, component } = screens[screen]
    registerComponent(id, () => component, store, Provider)
  }
}

export default { startApp, registerScreens, userLoggedIn: startCaronaBoard }
