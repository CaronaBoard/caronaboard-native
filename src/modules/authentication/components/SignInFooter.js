import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './styles/SignInFooterStyles'
import { screens } from '../../../configuration/navigation/Screens'

export class SignInFooter extends Component {
  static propTypes = {
    email: PropTypes.string,
    navigator: PropTypes.object.isRequired
  }

  navigateTo = screen => () => this.props.navigator.push({screen})

  render () {
    return (
      <View>
        <View style={Styles.textRow}>
          <RkButton
            key='forgot-password-button'
            rkType='clear'
            onPress={this.navigateTo(screens.forgotPassword.id)}
          >
            <RkText rkType='header6'>Forgot your password?</RkText>
          </RkButton>
        </View>
        <View style={Styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton
            key='signup-button'
            rkType='clear'
            onPress={this.navigateTo(screens.signUp.id)}
          >
            <RkText rkType='header6'> Sign up now</RkText>
          </RkButton>
        </View>
      </View>
    )
  }
}
