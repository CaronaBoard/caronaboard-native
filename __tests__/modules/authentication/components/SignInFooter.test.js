import React from 'react'
import { shallow } from 'enzyme'
import { SignInFooter } from '../../../../src/modules/authentication/components/SignInFooter'
import { RkButton } from 'react-native-ui-kitten'
import { screens } from '../../../../src/navigation/Screens'

const props = {
  email: 'email',
  navigator: {
    push: jest.fn()
  }
}

describe('<SignInFooter />', () => {
  const wrapper = shallow(<SignInFooter {...props} />)
  const keyIsEqualTo = key => n => n.key() === key

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should navigate to password recovery screen', () => {
    const forgotPasswordButton = wrapper.find(RkButton).findWhere(keyIsEqualTo('forgot-password-button'))
    forgotPasswordButton.simulate('press')
    expect(props.navigator.push).toHaveBeenCalledWith({ screen: screens.forgotPassword.id })
  })

  it('should navigate to signup screen', () => {
    const forgotPasswordButton = wrapper.find(RkButton).findWhere(keyIsEqualTo('signup-button'))
    forgotPasswordButton.simulate('press')
    expect(props.navigator.push).toHaveBeenCalledWith({ screen: screens.forgotPassword.id })
  })
})
