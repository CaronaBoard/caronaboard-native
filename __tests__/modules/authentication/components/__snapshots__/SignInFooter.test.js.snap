// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInFooter /> Should have a snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <RkButton
      onPress={[Function]}
      rkType="clear"
    >
      <RkText
        rkType="header6"
      >
        Forgot your password?
      </RkText>
    </RkButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <RkText
      rkType="primary3"
    >
      Donâ€™t have an account?
    </RkText>
    <RkButton
      onPress={[Function]}
      rkType="clear"
    >
      <RkText
        rkType="header6"
      >
         Sign up now
      </RkText>
    </RkButton>
  </View>
</View>
`;
